CommonIncludes/ve_avx512.hpp

```diff
@@ -0,0 +1,1941 @@
+		return _mm512_mask_load_epi32(_mm512_setzero_si512(), mask, (int32_t const*)(source + e.value));
+	}
+	template<typename T>
+	RELEASE_INLINE int_vector load(partial_contiguous_tags<T> e, uint32_t const* source) {
+		auto mask = __mmask16((1 << e.subcount) - 1);
+		return _mm512_mask_load_epi32(_mm512_setzero_si512(), mask, (int32_t const*)(source + e.value));
+	}
+
+	template<typename T, typename U>
+	RELEASE_INLINE auto load(partial_contiguous_tags<T> e, U const* source) -> std::enable_if_t<sizeof(U) == 4, tagged_vector<U>> {
+		auto mask = __mmask16((1 << e.subcount) - 1);
+		return _mm512_mask_load_epi32(_mm512_setzero_si512(), mask, (int32_t const*)(source + e.value));
+	}
+
+
+	// CONTINUE
+	namespace detail {
+		template<typename T>
+		constexpr auto zero_is_null_wrapper(int) -> std::enable_if_t<T::zero_is_null_t, bool> { return true; }
+		template<typename T>
+		constexpr auto zero_is_null_wrapper(...) { return false; }
+	}
+
+	template<typename T>
+	RELEASE_INLINE int_vector load(contiguous_tags<T> e, int16_t const* source) {
+		auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+		return _mm512_cvtepi16_epi32(v);
+	}
+	template<typename T>
+	RELEASE_INLINE int_vector load(contiguous_tags<T> e, uint16_t const* source) {
+		auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+		return _mm512_cvtepu16_epi32(v);
+	}
+	template<typename T, typename U>
+	RELEASE_INLINE auto load(contiguous_tags<T> e, U const* source) -> std::enable_if_t<sizeof(U) == 2, tagged_vector<U>> {
+		if constexpr(!detail::zero_is_null_wrapper<U>(0)) {
+			auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+			return _mm512_cvtepi16_epi32(v);
+		} else {
+			auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+			return _mm512_cvtepu16_epi32(v);
+		}
+	}
+
+	template<typename T>
+	RELEASE_INLINE int_vector load(unaligned_contiguous_tags<T> e, int16_t const* source) {
+		auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+		return _mm512_cvtepi16_epi32(v);
+	}
+	template<typename T>
+	RELEASE_INLINE int_vector load(unaligned_contiguous_tags<T> e, uint16_t const* source) {
+		auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+		return _mm512_cvtepu16_epi32(v);
+	}
+	template<typename T, typename U>
+	RELEASE_INLINE auto load(unaligned_contiguous_tags<T> e, U const* source) -> std::enable_if_t<sizeof(U) == 2, tagged_vector<U>> {
+		if constexpr(!detail::zero_is_null_wrapper<U>(0)) {
+			auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+			return _mm512_cvtepi16_epi32(v);
+		} else {
+			auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
+			return _mm512_cvtepu16_epi32(v);
+		}
+	}
+
+	template<typename T>
+	RELEASE_INLINE int_vector load(partial_contiguous_tags<T> e, int16_t const* source) {
+		auto mask = __mmask16((1 << e.subcount) - 1);
+		auto v = _mm256_loadu_si256((const __m256i *)(source + e.value));
```

Comment:
I think that you could be using _mm256_maskz_loadu_epi16 here (and for a number of the other partial contiguous tags loads) which would save you the blend step